projects/cdk/directives/for-async/for-async.directive.ts

```diff
@@ -0,0 +1,61 @@
+import {
+    Directive,
+    Inject,
+    Input,
+    NgZone,
+    OnDestroy,
+    TemplateRef,
+    ViewContainerRef,
+} from '@angular/core';
+import {tuiZonefree} from '@taiga-ui/cdk/observables';
+import {TuiDestroyService} from '@taiga-ui/cdk/services';
+import {from, of} from 'rxjs';
+import {concatMap, delay, takeUntil} from 'rxjs/operators';
+
+@Directive({
+    selector: '[tuiForAsync][tuiForAsyncOf]',
+    providers: [TuiDestroyService],
+})
+export class TuiForAsyncDirective<T> implements OnDestroy {
+    @Input() tuiForAsyncTimeout = 10;
+
+    constructor(
+        @Inject(ViewContainerRef) private readonly view: ViewContainerRef,
+        @Inject(TemplateRef) private readonly template: TemplateRef<unknown>,
+        @Inject(NgZone) private readonly ngZone: NgZone,
+        @Inject(TuiDestroyService) private readonly destroy$: TuiDestroyService,
+    ) {}
+
+    @Input()
+    set tuiForAsyncOf(items: T[] | readonly T[] | null | undefined) {
+        this.cleanPreviousNodes();
+
+        if (Array.isArray(items)) {
+            this.asyncCreateNewNodes(items);
+        }
+    }
+
+    ngOnDestroy(): void {
+        this.view.clear();
+    }
+
+    private cleanPreviousNodes(): void {
+        if (this.view.length > 0) {
+            this.view.clear();
+        }
+    }
+
+    private asyncCreateNewNodes(items: T[] | readonly T[]): void {
+        from(items)
+            .pipe(
+                tuiZonefree(this.ngZone),
+                concatMap(item => of(item).pipe(delay(this.tuiForAsyncTimeout))),
```

Comment:
I believe it's a case for `ngOnChanges`, otherwise there's a race condition for updating array and delay.